lib/execution_engine2/utils/MongoUtil.py

```diff
@@ -0,0 +1,158 @@
+
+import logging
+from pymongo import MongoClient
+from pymongo.errors import ServerSelectionTimeoutError
+import subprocess
+import traceback
+from bson.objectid import ObjectId
+
+
+class MongoUtil:
+
+    @classmethod
+    def _start_local_service(self):
+        logging.info('starting local mongod service')
+
+        logging.info('running sudo service mongodb start')
+        pipe = subprocess.Popen("sudo service mongodb start", shell=True, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        stdout = pipe.communicate()
+        logging.info(stdout)
+
+        logging.info('running mongod --version')
+        pipe = subprocess.Popen("mongod --version", shell=True, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        stdout = pipe.communicate()
+        logging.info(stdout)
+
+    @classmethod
+    def _get_collection(self, mongo_host, mongo_port, mongo_database, mongo_collection,
+                        mongo_user=None, mongo_password=None, mongo_authmechanism='DEFAULT'):
+        """
+        connect Mongo server and return a collection
+        """
+
+        if mongo_user:
+            logging.info('mongo-user found in config file, configuring client for authentication using mech ' + str(mongo_authmechanism) )
+            my_client = MongoClient(mongo_host, mongo_port,
+                                    username=mongo_user, password=mongo_password,
+                                    authSource=mongo_database,
+                                    authMechanism=mongo_authmechanism)
+        else:
+            logging.info('no mongo-user found in config file, connecting without auth')
+            my_client = MongoClient(mongo_host, mongo_port)
+
+        try:
+            my_client.server_info()  # force a call to server
+        except ServerSelectionTimeoutError as e:
+            error_msg = 'Connot connect to Mongo server\n'
+            error_msg += 'ERROR -- {}:\n{}'.format(
+                            e,
+                            ''.join(traceback.format_exception(None, e, e.__traceback__)))
+            raise ValueError(error_msg)
+
+        # TODO: check potential problems. MongoDB will create the collection if it does not exist.
+        my_database = my_client[mongo_database]
+        my_collection = my_database[mongo_collection]
+
+        return my_collection
+
+    def __init__(self, config):
+        self.mongo_host = config['mongo-host']
+        self.mongo_port = int(config['mongo-port'])
+        self.mongo_database = config['mongo-database']
+        self.mongo_user = config['mongo-user']
+        self.mongo_pass = config['mongo-password']
+        self.mongo_authmechanism = config['mongo-authmechanism']
+
+        self.mongo_collection = config['mongo-collection']
+
+        try:
+            start_local = int(config.get('start-local-mongo', 1))
+        except Exception:
```

Comment:
I think we should throw the exception here

Reply:
👍